### DISCUSSION TITLE
How to know what scope and endpoint to use, for accessing service XYZ?

### DISCUSSION THREAD

**Discussion by ecormaksin** [AzureAD/microsoft-authentication-library-for-python#529] [Q&A] [answered]
I refered [this sample code](https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/5b135b2e82d0a8064b226b8ca3d2ff5959070b05/sample/confidential_client_secret_sample.py).

I want to let Azure Function to authenticate by Azure Active Directory.

I cannot understand how to access to Azure BLOB Storage from Azure Functions using MSAL.
How do I configure `scope` and `endpoint` ?

**Comment by rayluo** [comment, marked as answer]
> how to access to Azure BLOB Storage
> How do I configure scope and endpoint ?

Please refer to this [introduction of scopes (permissions)](https://learn.microsoft.com/en-us/azure/active-directory/develop/permissions-consent-overview). Specific scopes and endpoints are defined by the service that you are going to use, so you will need to find them from Azure BLOB storage's documentation.

Last but not the least, this kind of general usage question shall better be asked in StackOverflow, which is back by a broader community.
